File: Week 1 (Engineering Concepts)/Algorithms_Data Structures/Code/SortingCustomerOrders/SortingOrder.java
public class SortingOrder {
    static class Order {
        int orderId;
        String customerName;
        double totalPrice;

        public Order(int orderId, String customerName, double totalPrice) {
            this.orderId = orderId;
            this.customerName = customerName;
            this.totalPrice = totalPrice;
        }

        public String toString() {
            return "Order ID: " + orderId + ", Name: " + customerName + ", Total: Rs." + totalPrice;
        }
    }

    static class OrderSorter {
        public void bubbleSort(Order[] orders) {
            int n = orders.length;
            for (int i = 0; i < n - 1; i++) {
                for (int j = 0; j < n - i - 1; j++) {
                    if (orders[j].totalPrice > orders[j + 1].totalPrice) {
                        Order temp = orders[j];
                        orders[j] = orders[j + 1];
                        orders[j + 1] = temp;
                    }
                }
            }
        }

        public void quickSort(Order[] orders, int low, int high) {
            if (low < high) {
                int pi = partition(orders, low, high);
                quickSort(orders, low, pi - 1);
                quickSort(orders, pi + 1, high);
            }
        }

        private int partition(Order[] orders, int low, int high) {
            double pivot = orders[high].totalPrice;
            int i = low - 1;
            for (int j = low; j < high; j++) {
                if (orders[j].totalPrice <= pivot) {
                    i++;
                    Order temp = orders[i];
                    orders[i] = orders[j];
                    orders[j] = temp;
                }
            }
            Order temp = orders[i + 1];
            orders[i + 1] = orders[high];
            orders[high] = temp;
            return i + 1;
        }
    }

    public static void main(String[] args) {
        Order[] orders = {
            new Order(101, "Alice", 1500.0),
            new Order(102, "Bob", 3000.0),
            new Order(103, "Charlie", 2000.0)
        };

        OrderSorter sorter = new OrderSorter();

        System.out.println("Before Sorting (Bubble):");
        for (Order o : orders) System.out.println(o);

        sorter.bubbleSort(orders);
        System.out.println("\nAfter Bubble Sort (Low to High):");
        for (Order o : orders) System.out.println(o);

        Order[] quickSortOrders = {
            new Order(101, "Alice", 1500.0),
            new Order(102, "Bob", 3000.0),
            new Order(103, "Charlie", 2000.0)
        };

        sorter.quickSort(quickSortOrders, 0, quickSortOrders.length - 1);
        System.out.println("\nAfter Quick Sort (Low to High):");
        for (Order o : quickSortOrders) System.out.println(o);
    }
}

File: Week-1(Engineering Concepts)/AlgorithmandDS/CODE/SortingOrder/SortingOrder/SortingOrder.cs
using System;

class Order
{
    public int Id;
    public string Name;
    public double Total;
}

class Sorter
{
    public static void Bubble(Order[] arr)
    {
        int n = arr.Length;
        for (int i = 0; i < n - 1; i++)
            for (int j = 0; j < n - i - 1; j++)
                if (arr[j].Total > arr[j + 1].Total)
                    (arr[j], arr[j + 1]) = (arr[j + 1], arr[j]);
    }
    public static void Quick(Order[] arr, int low, int high)
    {
        if (low < high)
        {
            int pi = Partition(arr, low, high);
            Quick(arr, low, pi - 1);
            Quick(arr, pi + 1, high);
        }
    }
    static int Partition(Order[] arr, int low, int high)
    {
        double pivot = arr[high].Total;
        int i = low - 1;
        for (int j = low; j < high; j++)
        {
            if (arr[j].Total <= pivot)
                (arr[++i], arr[j]) = (arr[j], arr[i]);
        }
        (arr[i + 1], arr[high]) = (arr[high], arr[i + 1]);
        return i + 1;
    }
}

class Program
{
    static void Main()
    {
        Order[] orders = {
            new Order{Id=1, Name="A", Total=300},
            new Order{Id=2, Name="B", Total=100},
            new Order{Id=3, Name="C", Total=200}
        };
        Order[] bubbleOrders = (Order[])orders.Clone();
        Sorter.Bubble(bubbleOrders);
        Console.WriteLine("Bubble Sort:");
        foreach (var o in bubbleOrders)
            Console.WriteLine($"{o.Name} - {o.Total}");
        Sorter.Quick(orders, 0, orders.Length - 1);
        Console.WriteLine("Quick Sort:");
        foreach (var o in orders)
            Console.WriteLine($"{o.Name} - {o.Total}");
    }
}
